README.md

```diff
@@ -28,25 +28,70 @@ public class ExampleIntegrationTest {
 
 See [Vert.x Config overloading rules](https://vertx.io/docs/vertx-config/java/#_overloading_rules) 
 for more details.
 
+#### KnotxConcatConfigProcessor
+
+This implementation of Vert.x [ConfigProcessor](https://vertx.io/docs/vertx-config/java/#_extending_the_config_retriever)
+enables KnotxExtension to load Knot.x configuration files in a true hierarchical manner,
+including cross-file references to variables (only with HOCON files).
+ 
+From a user standpoint, HOCON cross-references should be available out-of-box in Vert.x's processing of configurations, since when you load configs
+programatically, you provide a source for given file and its format, and only JSON and HOCON are supported so it would be quite logical assumption.  
+This is not the case, however, and what Vert.x does internally is load all given files independently, evaluate them independently of each other,
+and only after that's been done, stiches all files together to create a final result that gets served to Knot.x for processing.
+
+KnotxConcatConfigProcessor works around this problem. It creates a new config format, in which you specify a JSON file:
+
+```json
+{ 
+  "paths": [ "base.conf", "specific.json" ], 
+  "overrides": {"baseKey": "newValue"}
+}
+```
+
+Files from `paths` are loaded in given order, and `overrides` are applied directly on top of resulting HOCON Config object, and only then
+the configuration gets resolved and effectively returned for Knot.x for processing.
+
+Please refer to [HOCON readme](https://github.com/lightbend/config/blob/master/README.md) if you have any more questions regarding config loading
+behavior.
+
 ### KnotxWiremockExtension
 
-Standalone WireMock server injection and lifecycle management, currently allows for specifying on which port WireMockServer instance should be present*.
+Standalone WireMockServer injection and lifecycle management. Allows for:
+ 
+- Specifying on which port WireMockServer instance should be present,
+- If no port is specified on annotation, a random one will be assigned to given instance,
+- Running multiple mocked server instances,
+- Autoinjecting mocked servers' port numbers into Knot.x configuration (more details below).
+
+***Warning:*** if you use KnotxExtension, you **must not inject** KnotxWiremockExtension, 
+as the functionality of the latter gets auto-imported into the former.
 
-**Warning:** if you use KnotxExtension, you *must not inject* KnotxWiremockExtension, as the functionality of the latter is included in the former.
+When used through KnotxExtension, WireMockServer instances' ports will be available 
+for referencing in configuration under `test.wiremock.<mock_name>.port` variables (using HOCON syntax).  
+Server's `mock_name` for injecting into configuration is currently taken from class variable or parameter name.
+Mock name is an unique identifier, so if two different test classes specify the same mock name, only one mock instance 
```

Comment:
Please explain this scenario: two different test classes specify the same mock name. As I understand mock is configued per class.

Reply:
Test class name is not taken into account when creating mock. I'll change that since writing docs will be more problematic than a change in mock creation code.